menas/src/main/scala/za/co/absa/enceladus/menas/controllers/RestExceptionHandler.scala

```diff
@@ -40,6 +41,11 @@ class RestExceptionHandler {
 
   private val logger = LoggerFactory.getLogger(this.getClass)
 
+  @ExceptionHandler(value = Array(classOf[AsyncRequestTimeoutException]))
+  def handleAsyncRequestTimeoutException(exception: AsyncRequestTimeoutException): ResponseEntity[Any] = {
+    ResponseEntity.status(HttpStatus.BANDWIDTH_LIMIT_EXCEEDED).build[Any]()
```

Comment:
Guys, what's your opinion on this status code? This was something I wasn't sure about.

Reply:
Do we want to keep it as 509, server issue, we should have scaled better, or do we want to blame the people with 429, too many requests?

I am fine with both.

Reply:
The server code it is. AsyncTimeoutExpired is more server related. It is something unexpected rather than something we deliberately restrict.

Reply:
I would not blame an individual user (4xx), unless absolutely certain that they are at fault. 

Since the nature of the exception does not guarantee that (both that such a timeout may have generally been caused by the user only indirectly or not at all (another user's action or another "process" being run)), I would rather stick 5xx error code.

I'd personally use 500/503 (rather standardized with an explanatory message), but 509 with a message may not be any worse.

Reply:
Agree, 503 is much better fits the purpose.

We try not to use 500 explicitly so we know if 500 occurs, that means something went wrong, but are not sure what.